dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleAction.java

```diff
@@ -322,12 +334,29 @@ public void setDataElement(DataElement dataElement) {
 
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
+  @Property(required = Property.Value.FALSE, owner = Property.Value.TRUE)
   public String getTemplateUid() {
-    return templateUid;
+    return notificationTemplate != null ? notificationTemplate.getUid() : null;
+  }
+
+  public void setTemplateUid(String templateUid) {
+    if (templateUid == null) {
+      notificationTemplate = null;
+      return;
+    }
+
+    this.templateUid = templateUid;
+    ProgramNotificationTemplate template = new ProgramNotificationTemplate();
```

Comment:
Is there a possibility that creating a new `ProgramNotificationTemplate` here and only setting its `UID` could override an existing `ProgramNotificationTemplate` with the same `UID`, which already had an existing valid `messageTemplate` for example? (when objects are saved to the DB)
So `messageTemplate` could have been something before & now it has been overridden with `null`?
A test would confirm whether this is a potential issue or not (if that condition is not already checked)

Reply:
that test would also confirm whether the API is working as expected after the changes introduced, specifically the `ProgramRuleAction` `get`/`set` `templateUid` (can't see any API tests in the PR, unless existing tests cover the change)